src/settings/value.rs

```diff
@@ -72,6 +72,22 @@ pub enum Error {
 pub type Result<T> = StdResult<T, Error>;
 
 impl Value {
+    /// Decodes the value from JSON.
+    pub fn from_json<R>(reader: R) -> serde_json::Result<Self>
+    where
+        R: std::io::Read,
+    {
+        serde_json::from_reader(reader)
+    }
+
+    /// Encodes the value as JSON.
+    pub fn write_json<W>(&self, writer: W) -> serde_json::Result<()>
+    where
+        W: std::io::Write,
+    {
+        serde_json::to_writer(writer, self)
+    }
+
```

Comment:
This is a bit unrelated but I feel like a derive macro to automatically add these would be extremely useful for various places in the codebase.